### ISSUE TITLE
Add (optional) BIP148 mode

### ISSUE THREAD

**Issue by kek-coin** [voisine/breadwallet#425]
It should be possible to enforce BIP148 even with SPV wallets, as BIP148 only enforces versionbit signalling and therefore should be viable in header-chain mode.

**Comment by dabura667** [comment]
If you want to support BIP148 run your own full node. SPV can do nothing but follow the longest chain of block headers it is given.

If you think you can code a BIP148 mode into breadwallet with 0 bugs or problems for BIP148 supporters using such a codebase, then code your own and distribute it if you want.

Trying to get through a hard fork on an SPV wallet is irresponsible. Run a node.

**Comment by kek-coin** [comment]
> SPV can do nothing but follow the longest chain of block headers it is given.

Incorrect, BIP148 enforces things on block-header level so can be enforced in header-only mode.

**Comment by dabura667** [comment]
> so can be enforced in header-only mode.

Specifically how?

**Comment by kek-coin** [comment]
Three must be some logic for validating headers (e.g. regarding the necessary difficulty), BIP148 enforces limitations on the versionbits (must be signalling for segwit) until segwit is locked in/active (which is also a condition that can be derived from past headers).
